cnd/src/http_api/routes.rs

```diff
@@ -99,25 +100,108 @@ pub async fn handle_get_halight_swap(
                 state.fund_action().map(|_| "fund"),
                 state.redeem_action().map(|_| "redeem"),
             ];
-
-            make_swap_entity(swap_id, maybe_action_names)
+            make_swap_entity(swap_id, state, maybe_action_names)
         }
-    };
-
-    Ok(entity)
+    }
 }
 
-fn make_swap_entity(swap_id: LocalSwapId, maybe_action_names: Vec<Option<&str>>) -> siren::Entity {
-    let swap = siren::Entity::default().with_class_member("swap");
+fn make_swap_entity<S>(
+    swap_id: LocalSwapId,
+    state: S,
+    maybe_action_names: Vec<Option<&str>>,
+) -> anyhow::Result<siren::Entity>
+where
+    S: Status + MyRole + QuantityWei + QuantitySatoshi,
+{
+    let role = state.my_role();
+    let swap = SwapResource {
+        status: state.status(),
+        role: Http(role),
+    };
 
-    maybe_action_names
+    let mut entity = siren::Entity::default()
+        .with_class_member("swaps")
+        .with_properties(swap)
+        .map_err(|e| {
+            tracing::error!("failed to set properties of entity: {:?}", e);
+            HttpApiProblem::with_title_and_type_from_status(StatusCode::INTERNAL_SERVER_ERROR)
+        })?
+        .with_link(siren::NavigationalLink::new(
+            &["self"],
+            route_factory::swap_path(swap_id),
+        ));
+
+    let alpha_params = HanEthereum {
+        protocol: "han-ethereum".to_string(),
+        quantity: state.quantity_wei(),
+    };
+    let alpha_params_sub = siren::SubEntity::from_entity(
+        siren::Entity::default()
+            .with_class_member("parameters")
+            .with_properties(alpha_params)
+            .map_err(|e| {
+                tracing::error!("failed to set properties of entity: {:?}", e);
+                HttpApiProblem::with_title_and_type_from_status(StatusCode::INTERNAL_SERVER_ERROR)
+            })?,
```

Comment:
```suggestion
            .context("failed to set alpha_params as properties for sub-entity")?,
```

Reply:
This suggestion doesn't build?

Reply:
You should be able to call `context` on anything that returns a Result where the `Error` implements `std::error::Error`.

Did I make an epic fail and return an error in the siren library that doesn't implement StdError?

Reply:
```
rs:47:1
    |
47  |   pub enum EntityBuilderError {
    |   --------------------------- doesn't satisfy `_: anyhow::context::ext::StdError`
    |
    = note: the method `context` exists but the following trait bounds were not satisfied:
            `siren_types::types::EntityBuilderError: anyhow::context::ext::StdError`
            which is required by `std::result::Result<siren_types::types::Entity, siren_types::types::EntityBuilderError>: anyhow::Context<siren_types::types::Entity, siren_types::types::EntityBuilderError>`
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
            `use quick_error::ResultExt;`
```

Reply:
LOL, I just went to raise an issue on the siren-types repo to find it was ours - face palm.

Reply:
Why is siren-types under CoBloX organisation and not comit-network?

Reply:
Raised an issue on siret-types repo to track this.  https://github.com/coblox/rust-siren-types/issues/5

Reply:
Because it doesn't really have anything to do with COMIT specifically, it is just a library :)

Reply:
Epic fail Thomas! Why does this not implement `std::error::Error`!